ipn/ipnlocal/local_test.go

```diff
@@ -922,41 +923,191 @@ func TestWireguardExitNodeDNSResolvers(t *testing.T) {
 		nm := &netmap.NetworkMap{}
 		gotResolvers, gotOK := wireguardExitNodeDNSResolvers(nm, peers, tc.id)
 
-		if gotOK != tc.wantOK || !resolversEqual(gotResolvers, tc.wantResolvers) {
+		if gotOK != tc.wantOK || !resolversEqual(t, gotResolvers, tc.wantResolvers) {
 			t.Errorf("case: %s: got %v, %v, want %v, %v", tc.name, gotOK, gotResolvers, tc.wantOK, tc.wantResolvers)
 		}
 	}
 }
 
-func TestDNSConfigForNetmapForWireguardExitNode(t *testing.T) {
-	resolvers := []*dnstype.Resolver{{Addr: "dns.example.com"}}
-	nm := &netmap.NetworkMap{}
-	peers := map[tailcfg.NodeID]tailcfg.NodeView{
-		1: (&tailcfg.Node{
+func TestDNSConfigForNetmapForExitNodeConfigs(t *testing.T) {
+	type tc struct {
+		name                 string
+		exitNode             tailcfg.StableNodeID
+		peers                []tailcfg.NodeView
+		dnsConfig            *tailcfg.DNSConfig
+		wantDefaultResolvers []*dnstype.Resolver
+		wantRoutes           map[dnsname.FQDN][]*dnstype.Resolver
+	}
+
+	defaultResolvers := []*dnstype.Resolver{{Addr: "default.example.com"}}
+	wgResolvers := []*dnstype.Resolver{{Addr: "wg.example.com"}}
+	peers := []tailcfg.NodeView{
+		(&tailcfg.Node{
 			ID:                   1,
-			StableID:             "1",
+			StableID:             "wg",
 			IsWireGuardOnly:      true,
-			ExitNodeDNSResolvers: resolvers,
+			ExitNodeDNSResolvers: wgResolvers,
 			Hostinfo:             (&tailcfg.Hostinfo{}).View(),
 		}).View(),
+		// regular tailscale exit node with DNS capabilities
+		(&tailcfg.Node{
+			Cap:      26,
+			ID:       2,
+			StableID: "ts",
+			Hostinfo: (&tailcfg.Hostinfo{}).View(),
+		}).View(),
 	}
-	prefs := &ipn.Prefs{
-		ExitNodeID: "1",
-		CorpDNS:    true,
+	exitDOH := peerAPIBase(&netmap.NetworkMap{Peers: peers}, peers[0]) + "/dns-query"
+	routes := map[dnsname.FQDN][]*dnstype.Resolver{
+		"route.example.com.": {{Addr: "route.example.com"}},
+	}
+	stringifyRoutes := func(routes map[dnsname.FQDN][]*dnstype.Resolver) map[string][]*dnstype.Resolver {
+		if routes == nil {
+			return nil
+		}
+		m := make(map[string][]*dnstype.Resolver)
+		for k, v := range routes {
+			m[string(k)] = v
+		}
+		return m
 	}
 
-	got := dnsConfigForNetmap(nm, peers, prefs.View(), t.Logf, "")
-	if !resolversEqual(got.DefaultResolvers, resolvers) {
-		t.Errorf("got %v, want %v", got.DefaultResolvers, resolvers)
+	tests := []tc{
+		{
+			name:                 "no exit node, no routes, no resolver",
```

Comment:
nit:
use `/` instead of ` `
so like
`noExit/noRoutes/noResolver`
`tsExit/noRoutes/noResolver`